homeassistant/components/integration/sensor.py

```diff
@@ -112,9 +120,12 @@ def __init__(
             self._unit_of_measurement = None
         else:
             self._unit_of_measurement = unit_of_measurement
+            self._set_attributes()
 
         self._unit_prefix = UNIT_PREFIXES[unit_prefix]
         self._unit_time = UNIT_TIME[unit_time]
+        self._attr_last_reset = dt_util.utc_from_timestamp(0)
```

Comment:
We need to restore last reset as well

Reply:
Is that needed when it is constant?

Reply:
we are putting too much energy logic into this integration....

Reply:
> Is that needed when it is constant?

It isn't constant. As it restores (which can fail, thus, is a reset).
So I would expect this to set the current time set as a default (like state is set to 0), overwrite the attribute with the restored value together with the state value that is being restored.

That way, it resets nicely when restoring fails (or when first put to use).

> we are putting too much energy logic into this integration....

Well, energy is a much-used current example due to the release, I'm happy to talk about 3m gas in my further communication? Doesn't change a thing about this PR or comments.

Reply:
my point is that last_reset is not a concept associated to a generic integration such as this one.

little bit of history: the code in this integration initially was part of utility_meter, but was split exactly to make it generic

Reply:
> my point is that last_reset is not a concept associated to a generic integration such as this one.

It is actually. No matter what is being integrated here, it still has the statistical value, which is what `last_reset` powers.